AssertionError: call_args and arg_types do not match

Repro:
```
TORCHINDUCTOR_ABI_COMPATIBLE=1 TORCHINDUCTOR_CPP_WRAPPER=1 python test/inductor/test_torchinductor.py -k GPUTests.test_consecutive_split_cumprod_cuda
```
Error:
```
  File "/data/users/binbao/pytorch/torch/_inductor/graph.py", line 1544, in codegen_with_cpp_wrapper
    compiled = self.compile_to_module().call
  File "/data/users/binbao/pytorch/torch/_dynamo/utils.py", line 246, in time_wrapper
    r = func(*args, **kwargs)
  File "/data/users/binbao/pytorch/torch/_inductor/graph.py", line 1678, in compile_to_module
    self.codegen_with_cpp_wrapper() if self.cpp_wrapper else self.codegen()
  File "/data/users/binbao/pytorch/torch/_inductor/graph.py", line 1633, in codegen
    self.scheduler.codegen()
  File "/data/users/binbao/pytorch/torch/_dynamo/utils.py", line 246, in time_wrapper
    r = func(*args, **kwargs)
  File "/data/users/binbao/pytorch/torch/_inductor/scheduler.py", line 2827, in codegen
    self.get_backend(device).codegen_node(node)
  File "/data/users/binbao/pytorch/torch/_inductor/codegen/cuda_combined_scheduling.py", line 81, in codegen_node
    return self._triton_scheduling.codegen_node(node)
  File "/data/users/binbao/pytorch/torch/_inductor/codegen/simd.py", line 1166, in codegen_node
    return self.codegen_node_schedule(node_schedule, buf_accesses, numel, rnumel)
  File "/data/users/binbao/pytorch/torch/_inductor/codegen/simd.py", line 1384, in codegen_node_schedule
    final_kernel.call_kernel(final_kernel.kernel_name)
  File "/data/users/binbao/pytorch/torch/_inductor/codegen/triton.py", line 2751, in call_kernel
    wrapper.generate_kernel_call(
  File "/data/users/binbao/pytorch/torch/_inductor/codegen/wrapper.py", line 1617, in generate_kernel_call
    assert arg_types is not None and len(call_args) == len(
torch._dynamo.exc.BackendCompilerFailed: backend='compile_fx_wrapper' raised:
AssertionError: call_args and arg_types do not match
```

cc @chenyang78 @penguinwu

--- a/test/inductor/test_cuda_cpp_wrapper.py
+++ b/test/inductor/test_cuda_cpp_wrapper.py
@@ -61,9 +61,6 @@ test_failures_cuda_wrapper = {
     "test_mm_plus_mm2_cuda_dynamic_shapes": test_torchinductor.TestFailure(
         ("cuda_wrapper",), is_skip=True
     ),
-    "test_consecutive_split_cumprod_cuda_dynamic_shapes": test_torchinductor.TestFailure(
-        ("cuda_wrapper",), is_skip=True
-    ),
 }
 
 

--- a/torch/_inductor/codegen/cpp_wrapper_cuda.py
+++ b/torch/_inductor/codegen/cpp_wrapper_cuda.py
@@ -440,13 +440,13 @@ class CppWrapperCuda(CppWrapperCpu):
         if config.triton.autotune_at_compile_time:
             self.kernel_autotune_calls.writeline(line)
         if zero_fill:
-            # TODO: remove this function to use the default WrapperCodegen behavior after service platform has zero_() symbol
-            # default behavior is f"workspace.zero_(){self.ending}"
             if isinstance(nbytes, sympy.Expr):
                 self.writeline(
                     f"workspace.zero_(){self.ending}"
                 )
             else:
+                # TODO: remove this function to use the default WrapperCodegen behavior after service platform has zero_() symbol
+                # default behavior is f"workspace.zero_(){self.ending}"
                 self.writeline(
                     f"AOTI_TORCH_ERROR_CODE_CHECK(aoti_torch_zero_(workspace.get())){self.ending}"
                 )

--- a/torch/_inductor/codegen/cpp_wrapper_cuda.py
+++ b/torch/_inductor/codegen/cpp_wrapper_cuda.py
@@ -442,8 +442,13 @@ class CppWrapperCuda(CppWrapperCpu):
         if zero_fill:
             # TODO: remove this function to use the default WrapperCodegen behavior after service platform has zero_() symbol
             # default behavior is f"workspace.zero_(){self.ending}"
-            self.writeline(
-                f"AOTI_TORCH_ERROR_CODE_CHECK(aoti_torch_zero_(workspace.get())){self.ending}"
-            )
+            if isinstance(nbytes, sympy.Expr):
+                self.writeline(
+                    f"workspace.zero_(){self.ending}"
+                )
+            else:
+                self.writeline(
+                    f"AOTI_TORCH_ERROR_CODE_CHECK(aoti_torch_zero_(workspace.get())){self.ending}"
+                )
             if config.triton.autotune_at_compile_time:
                 self.kernel_autotune_calls.writeline(f"workspace.zero_(){self.ending}")

--- a/torch/_inductor/codegen/cpp_wrapper_cuda.py
+++ b/torch/_inductor/codegen/cpp_wrapper_cuda.py
@@ -441,9 +441,7 @@ class CppWrapperCuda(CppWrapperCpu):
             self.kernel_autotune_calls.writeline(line)
         if zero_fill:
             if isinstance(nbytes, sympy.Expr):
-                self.writeline(
-                    f"workspace.zero_(){self.ending}"
-                )
+                self.writeline(f"workspace.zero_(){self.ending}")
             else:
                 # TODO: remove this function to use the default WrapperCodegen behavior after service platform has zero_() symbol
                 # default behavior is f"workspace.zero_(){self.ending}"

--- a/torch/_inductor/codegen/wrapper.py
+++ b/torch/_inductor/codegen/wrapper.py
@@ -1490,6 +1490,8 @@ class WrapperCodeGen(CodeGen):
         return SymbolicCallArg(expr, tree.numel)
 
     def generate_workspace_allocation(self, nbytes, device, zero_fill):
+        if isinstance(nbytes, sympy.Expr):
+            nbytes = V.graph.sizevars.size_hint(nbytes)
         line = self.make_allocation(
             "workspace", device, torch.uint8, shape=(nbytes,), stride=(1,)
         )

